environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/timeout/NotificationTimeoutTest.java

```diff
@@ -0,0 +1,197 @@
+                .fireAsync(new CountDownLatch(0), NotificationOptions.of(NotificationOptionKeys.TIMEOUT, 1.2345));
+        }
+    }
+
+    @Test
+    public void testTimeoutNotReached() throws InterruptedException {
+        // this deployment omits the SLOW observer so as to AVOID timeout - e.g. this deployment has only one observer
+        final String success = "Success";
+        Weld weld = new Weld().disableDiscovery().addBeanClasses(HardworkingObserver.class);
+        try (WeldContainer container = weld.initialize()) {
+            BlockingQueue<String> synchronizer = new LinkedBlockingQueue<>();
+            CountDownLatch countdown = new CountDownLatch(1);
+            container.event().select(CountDownLatch.class)
+                .fireAsync(countdown, NotificationOptions.of(NotificationOptionKeys.TIMEOUT, new Long(2000l)))
+                .thenRun(() -> synchronizer.add(success))
+                .exceptionally((Throwable t) -> { // only executes if the notification ended with exception
+                    synchronizer.add("Exception");
+                    return null;
+                });
+            // wait for the observer to finish the job
+            countdown.await();
+            // assert that BlockingQueue has exactly one item indicating success
+            String fromSynchronizer = synchronizer.poll(3, TimeUnit.SECONDS);
+            Assert.assertNotNull(fromSynchronizer);
+            Assert.assertTrue(synchronizer.isEmpty());
+            Assert.assertEquals(success, fromSynchronizer);
+        }
+    }
+
+    @Test
+    public void testSingleObserverNotificationTimeout() throws InterruptedException {
+        SUCCESSION_OF_EVENTS.clear();
+        // this deployment omits the FAST observer so as to FORCE timeout - e.g. this deployment has only one observer
+        Weld weld = new Weld().disableDiscovery().addBeanClasses(LazyObserver.class);
+        try (WeldContainer container = weld.initialize()) {
+            BlockingQueue<Throwable> synchronizer = new LinkedBlockingQueue<>();
+            CountDownLatch countdown = new CountDownLatch(1);
+            container.event().select(CountDownLatch.class)
+                .fireAsync(countdown, NotificationOptions.of(NotificationOptionKeys.TIMEOUT, new Long(1000l)))
+                .exceptionally((Throwable t) -> {
+                    SUCCESSION_OF_EVENTS.add("Timeout");
+                    synchronizer.add(t);
+                    return null;
+                });
+            Throwable fromSynchronizer = synchronizer.poll(5, TimeUnit.SECONDS);
+            Assert.assertNotNull(fromSynchronizer);
+            Assert.assertTrue(fromSynchronizer.getCause().toString().contains(TimeoutException.class.getSimpleName()));
+            // wait for the observer to finish the jobs
+            countdown.await();
+            // assert that two events happened and timeout came first
+            Assert.assertTrue(SUCCESSION_OF_EVENTS.size() == 2);
```

Comment:
This assertion fails when you execute the whole test class, but passes when you execute just this method.
And I am blind and cannot see why, will take a look at it tomorrow.